### DISCUSSION TITLE
part-ii/2020-103/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#309] [Question comments]
Part II 2020 Number Theory: Paper 2, Section I, $1 \mathbf{H}$ 

https://questions.tripos.org/part-ii/2020-103/

**Comment by callumhobbis** [comment]
<details>
<summary>Sketch solution</summary>

All bookwork with a standard method at the end.

The $(-1)^{n-1}$ relation is done by induction. The next part can be proved via its contrapositive - then it becomes a corollary of the result "for $p,q\in\mathbb Z$ with $0<q<q_{n+1}$, $|q\theta-p|\geq|q_n\theta-p_n|$".

<details>
<summary>Waffling about this proof</summary>

**Proof in lectures:** This result uses the sneaky trick that $\exists u,v\in\mathbb Z$ with $p=up_n+vp_{n+1}$ and $q=uq_n+vq_{n+1}$, as this comes from a matrix equation with determinant $p_nq_{n-1}-p_{n-1}q_n$. Since this is $\pm1$, we're not dividing by anything, so $u,v$ are indeed _integers_. We substitute these into $q\theta-p$, and note if $v=0$ we're done. If not, we do some trickery with "$u,v$ have opposite signs, and these things have opposite signs, so their product have the _same_ signs, so we can split the absolute value as a sum, etc etc" to get it.

</details>

For the last part, I got $\sqrt{12}=[3,\overline{2,6}]$ and solution $(x,y)=(5,2)$.

</details>

**Reply by walrus1729** [reply]
<details>
<summary>Solution to last part</summary>

The continued fraction expansion of $\sqrt{12}$ is indeed $[3,\overline{2,6}]$, but $(5,2)$ is not a solution; our first two convergents are in fact $\frac{3}{1}$ and $\frac{7}{2}$, and the second of these gives the solution $(x,y) = (7,2)$.

</details>
